### PR TITLE
Show weird join field error.

### PR DESCRIPTION
**Pull request by TobiasNx**
The subfield $a only exists once. But the join_field behaves like there are multiple subfields $a.

### CODE REVIEW

File: src/main/resources/alma/fix/relatedRessourcesAndLinks.fix [hunk 1]
```diff
@@ -152,13 +156,19 @@ do list(path: "830??", "var": "$i")
       copy_field("$i.w", "isPartOf[].$last.hasSuperordinate[].$last.id")
       replace_all("isPartOf[].$last.hasSuperordinate[].$last.id", "^\\(DE-605\\)(.*)$", "http://lobid.org/resources/$1#!")
     end
-    copy_field("$i.a", "isPartOf[].$last.hasSuperordinate[].$last.label")
+    set_array("isPartOf[].$last.hasSuperordinate[].$last.label", " / ")
+    do list(path:"$i.a", "var":"$j")
+      copy_field("$j", "isPartOf[].$last.hasSuperordinate[].$last.label.$append")
+    end
+    join_field("isPartOf[].$last.hasSuperordinate[].$last.label", " / ")
```
Comment (TobiasNx) [inline src/main/resources/alma/fix/relatedRessourcesAndLinks.fix:163]: The subfield $a only exists once. But the join_field behaves like there are multiple subfields $a.

File: src/test/resources/alma-fix/990058434730206441.json [hunk 2]
```diff
@@ -70,7 +70,7 @@
     "type" : [ "IsPartOfRelation" ],
     "hasSuperordinate" : [ {
       "id" : "http://lobid.org/resources/HT001247609#!",
-      "label" : "Studia Judaica"
+      "label" : " /  / Studia Judaica"
```
Comment (TobiasNx) [inline src/test/resources/alma-fix/990058434730206441.json:73]: See output here

File: src/test/resources/alma-fix/990059571560206441.json [hunk 3]
```diff
@@ -70,7 +70,7 @@
     "type" : [ "IsPartOfRelation" ],
     "hasSuperordinate" : [ {
       "id" : "http://lobid.org/resources/HT002156469#!",
-      "label" : "DIN-Taschenbuch"
+      "label" : " /  / DIN-Taschenbuch"
```
Comment (TobiasNx) [inline src/test/resources/alma-fix/990059571560206441.json:73]: Or here

**Comment by fsteeg** [comment]
Not sure about the context, but could something like `uniq("isPartOf[].$last.hasSuperordinate[].$last.label")` before joining help here? Or setting `$last` instead of `$append` in `copy_field`?

**Comment by TobiasNx** [comment]
It might be this problem: https://github.com/metafacture/metafacture-fix/issues/261

File: src/main/resources/alma/fix/relatedRessourcesAndLinks.fix [hunk 4]
```diff
@@ -152,13 +156,19 @@ do list(path: "830??", "var": "$i")
       copy_field("$i.w", "isPartOf[].$last.hasSuperordinate[].$last.id")
       replace_all("isPartOf[].$last.hasSuperordinate[].$last.id", "^\\(DE-605\\)(.*)$", "http://lobid.org/resources/$1#!")
     end
-    copy_field("$i.a", "isPartOf[].$last.hasSuperordinate[].$last.label")
+    set_array("isPartOf[].$last.hasSuperordinate[].$last.label", " / ")
```
Comment (blackwinter) [inline src/main/resources/alma/fix/relatedRessourcesAndLinks.fix:159]: Is this intentional, setting the array to `" / "`?

File: src/main/resources/alma/fix/relatedRessourcesAndLinks.fix (hunk 1 above)
Comment (blackwinter) [reply src/main/resources/alma/fix/relatedRessourcesAndLinks.fix:163]: You're joining the initial value `" / "` and the subfield value `"Studia Judaica"` with your separator `" / "`, so the output `" /  / Studia Judaica"` is correct, isn't it?

To illustrate the individual parts: `"(?<initial value> / )(?<separator> / )(?<subfield value>Studia Judaica)"`

File: src/main/resources/alma/fix/relatedRessourcesAndLinks.fix (hunk 4 above)
Comment (blackwinter) [reply src/main/resources/alma/fix/relatedRessourcesAndLinks.fix:159]: Also, are you intentionally not `$append`ing the label here?

Comment (TobiasNx) [reply src/main/resources/alma/fix/relatedRessourcesAndLinks.fix:159]: Thanks, that was an dumb error!

**Pull request closed without merging**